docs/changelog.rst

```diff
@@ -21,11 +21,47 @@ Changes:
   ``dec_ref()`` are now enabled by default again.
   `#4246 <https://github.com/pybind/pybind11/pull/4246>`_
 
+* ``py::initialize_interpreter()`` using ``PyConfig_InitPythonConfig()``
+  instead of ``PyConfig_InitIsolatedConfig()``, to obtain complete
+  ``sys.path``.
+  `#4473 <https://github.com/pybind/pybind11/pull/4473>`_
+
+* Cast errors now always include Python type information, even if
+  ``PYBIND11_DETAILED_ERROR_MESSAGES`` is not defined. This increases binary
+  sizes slightly (~1.5%) but the error messages are much more informative.
+  `#4463 <https://github.com/pybind/pybind11/pull/4463>`_
+
+
 Build system improvements:
 
 * Update clang-tidy to 15 in CI.
   `#4387 <https://github.com/pybind/pybind11/pull/4387>`_
 
+* Moved the linting framework over to Ruff.
+  `#4483 <https://github.com/pybind/pybind11/pull/4483>`_
+
+Version 2.10.4 (Feb 24, 2023)
+----------------------------
+
+Changes:
+
+* ``python3 -m pybind11`` gained a ``--version`` option (prints the version and
+  exits), for feature parity between detection mechanisms.
```

Comment:
It would be nice to give a little bit of a clue what these "detection mechanisms" are.

My own very vague understanding: detection mechanisms in build systems inspecting available dependencies.

Or maybe we could add something like ` detection mechanisms (such as ...)`?

Reply:
I'd be fine to just drop it, the feature is simple enough to be understood without the extra text.